### PR TITLE
add missing known limitations

### PR DESCRIPTION
**Pull request by taroface**
During the process for identifying new 24.2 known limitations in SQL Queries, @rmloveland surfaced many that had been previously tagged but not yet documented. This PR should cover them all.

None of the known limitations turned out to be new in 24.2, so I've added them to both 24.1 and 24.2 docs.

### CODE REVIEW

**Review by DrewKimball — APPROVED** [review]
LGTM - Just had a few minor comments.

File: src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md
Comment (DrewKimball) [inline src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md:24]: Note: this is likely a Postgres bug, so maybe doesn't fit well as a CRDB "limitation". I'm not sure what the precedent is for a case like that, though.

File: src/current/_includes/v24.1/known-limitations/udf-limitations.md
Comment (DrewKimball) [inline src/current/_includes/v24.1/known-limitations/udf-limitations.md:10]: While the issue only mentions enum types, I believe this restriction applies to any user-defined type.

**Review by DrewKimball — APPROVED** [review]
Oops, forgot to flush the comments before.

**Review by rytaft — APPROVED** [review]
@DrewKimball is the expert here, but LGTM once you get his approval

File: src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md
Comment (taroface) [reply src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md:24]: I agree it's a stretch. In that case, I feel we should remove the `docs-known-limitation` label from #115680 and not document it here. I'll go ahead and remove it from this PR.

File: src/current/_includes/v24.2/known-limitations/plpgsql-limitations.md
Comment (florence-crl) [inline src/current/_includes/v24.2/known-limitations/plpgsql-limitations.md:22]: nit
```suggestion
- The `FOUND` local variable, which checks whether a statement affected any rows, is not supported in PL/pgSQL. [#122306](https://github.com/cockroachdb/cockroach/issues/122306)
```

File: src/current/_includes/v24.2/known-limitations/routine-limitations.md
Comment (florence-crl) [inline src/current/_includes/v24.2/known-limitations/routine-limitations.md:7]: btw, I could not find public docs for type `RECORD`.

**Review by florence-crl — APPROVED** [review]
lgtm with 1 non-blocking nit

File: src/current/_includes/v24.2/known-limitations/routine-limitations.md
Comment (taroface) [reply src/current/_includes/v24.2/known-limitations/routine-limitations.md:7]: @DrewKimball Is this specific to routines, or are we actually missing a `RECORD` data type page (like https://www.cockroachlabs.com/docs/stable/int etc.)?

**Pull request merged**
